### PR TITLE
O3-2252 Calculated values shouldn't be overwritten by the Encounter's…

### PR DESCRIPTION
**Pull request by CynthiaKamau**
… values

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR addresses the issue of null/undefined values on calculated fields in edit mode .
Calculated value is persisted when :
-There is a default value
-The encounter value is undefined
-The encounter value and calculated value are the same, if different

## Screenshots

https://www.loom.com/share/866edea9d4c34b91ab4ef3ef93eba49c?sid=7ffa7c0d-58cd-42e2-a30f-64d4448462d7

### CODE REVIEW

File: src/hooks/useInitialValues.ts [hunk 1]
```diff
@@ -77,10 +77,39 @@ export function useInitialValues(
...
+                // Assuming setInitialValues is a function that updates initialValues
+                if (
+                  res === existingVal ||
+                  (res && !existingVal) ||
+                  (res && field.questionOptions.defaultValue)
+                ) {
+                  setInitialValues((prevValues) => ({
+                    ...prevValues,
+                    [field.id]: res,
+                  }));
```
Comment (samuelmale) [inline src/hooks/useInitialValues.ts:102]: I would use a similar approach used [here](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/hooks/useInitialValues.ts#L113-L123) for handling async operations or implement a better approach. If the calculated value is empty, we should fallback to picking the value from the encounter.

File: src/hooks/useInitialValues.ts [hunk 2]
```diff
@@ -77,10 +77,39 @@ export function useInitialValues(
...
+                }
+              })
+              .catch((err) => {
+                initialValues[field.id] = err?.message;
```
Comment (samuelmale) [inline src/hooks/useInitialValues.ts:106]: Why would you initialise the field with an error message?

**Review by samuelmale — COMMENTED** [review]
Thanks for working on this @CynthiaKamau, two minor concerns:

- The ticket ID referenced in the PR title doesn't match the one used to name the branch. 
- Can we have some test coverage around this logic?

File: src/hooks/useInitialValues.ts (hunk 1 above)
Comment (CynthiaKamau) [reply src/hooks/useInitialValues.ts:102]: Since the evaluation is asynchronous , the returned value will always be a promise

**Pull request closed without merging**